### ISSUE TITLE
Python OverflowError when binaryview length is >= 0x8000000000000000

### ISSUE THREAD

**Issue by SiD3W4y** [Vector35/binaryninja-api#2081]
When a binaryview length is over 0x8000000000000000 it subtly breaks the python api (and prevents some plugins from being used). Whenever the `__len__` method is called it triggers a python OverflowError. This situation can occur in binaries such as coredumps where the mapping for vsyscall is present and vsyscall is at the far end of the address space.

![2020-10-31_621x824_scrot](https://user-images.githubusercontent.com/6969433/97776463-806b5a80-1b68-11eb-96f3-7529bc9049d0.png)

![2020-10-31_748x810_scrot](https://user-images.githubusercontent.com/6969433/97776471-9416c100-1b68-11eb-9eb4-f616a4c75721.png)

[binaryview_len_bug.zip](https://github.com/Vector35/binaryninja-api/files/5468859/binaryview_len_bug.zip)

**Comment by jeffli678** [comment]
Here is an explanation for what is happening: https://stackoverflow.com/a/4754427. 

It turns out that we should not use __len__ to return the length of the binary view. In Python, __len__ returns the number of elements in a collection, e.g., a list. The semantic is different here. Maybe we should just have a `length` attribute on BinaryView. We also need to check the whole Python API to look for similar issues.

**Comment by plafosse** [comment]
```>>> class Foo():
...  def __len__(self):
...   return 0x8000000000000000
...
>>> len(Foo())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: cannot fit 'int' into an index-sized integer
```

This is a limitation of Python and really has nothing to do with BinaryNinja. We need to deprecate the usage of `len(BinaryView)`

**Comment by psifertex** [comment]
I think besides any plugins, the only places we need to change are: 
API:
https://github.com/Vector35/binaryninja-api/blob/5435b69c1e109d275a2cff4e4249b6dea7c383b2/python/fileaccessor.py#L51-L54
https://github.com/Vector35/binaryninja-api/blob/dev/python/binaryview.py#L1365
https://github.com/Vector35/binaryninja-api/blob/dev/python/binaryview.py#L1416-L1422
https://github.com/Vector35/binaryninja-api/blob/dev/python/binaryview.py#L1445-L1452

Test related:
https://github.com/Vector35/binaryninja-api/blob/dev/suite/testcommon.py#L424
https://github.com/Vector35/binaryninja-api/blob/dev/suite/api_test.py#L160
https://github.com/Vector35/binaryninja-api/blob/dev/suite/api_test.py#L174

I would note that having `len()` on a raw file view probably /does/ make sense.

**Comment by CouleeApps** [comment]
For reference, Python decided this is a wontfix https://bugs.python.org/issue21444

**Comment by plafosse** [comment]
This issue is addressed in 2.4.3000

**Comment by xusheng6** [comment]
Fixed again https://github.com/Vector35/binaryninja-api/commit/683b07bba5adad52811c324725ebce783ad3ec84
